sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockService.java

```diff
@@ -0,0 +1,153 @@
+// SPDX-FileCopyrightText: 2022 2023 RTE FRANCE
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfenergy.compas.sct.commons;
+
+import org.apache.commons.lang3.StringUtils;
+import org.lfenergy.compas.scl2007b4.model.*;
+import org.lfenergy.compas.sct.commons.api.ControlBlockEditor;
+import org.lfenergy.compas.sct.commons.dto.*;
+import org.lfenergy.compas.sct.commons.dto.ControlBlockNetworkSettings.NetworkRanges;
+import org.lfenergy.compas.sct.commons.dto.ControlBlockNetworkSettings.RangesPerCbType;
+import org.lfenergy.compas.sct.commons.dto.ControlBlockNetworkSettings.Settings;
+import org.lfenergy.compas.sct.commons.dto.ControlBlockNetworkSettings.SettingsOrError;
+import org.lfenergy.compas.sct.commons.exception.ScdException;
+import org.lfenergy.compas.sct.commons.scl.SclRootAdapter;
+import org.lfenergy.compas.sct.commons.scl.ied.*;
+import org.lfenergy.compas.sct.commons.util.*;
+
+import java.util.*;
+import java.util.stream.Stream;
+
+public class ControlBlockService implements ControlBlockEditor {
+
+
+    @Override
+    public List<SclReportItem> analyzeDataGroups(SCL scd) {
+        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
+        return sclRootAdapter.streamIEDAdapters()
+                .map(iedAdapter -> {
+                    List<SclReportItem> list = new ArrayList<>();
+                    list.addAll(iedAdapter.checkDataGroupCoherence());
+                    list.addAll(iedAdapter.checkBindingDataGroupCoherence());
+                    return list;
+                }).flatMap(Collection::stream).toList();
+    }
+
+    @Override
+    public List<SclReportItem> createDataSetAndControlBlocks(SCL scd, Set<FcdaForDataSetsCreation> allowedFcdas) {
+        checkFcdaInitDataPresence(allowedFcdas);
+        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
+        Stream<LDeviceAdapter> lDeviceAdapters = sclRootAdapter.streamIEDAdapters().flatMap(IEDAdapter::streamLDeviceAdapters);
+        return createDataSetAndControlBlocks(lDeviceAdapters, allowedFcdas);
+    }
+
+    @Override
+    public List<SclReportItem> createDataSetAndControlBlocks(SCL scd, String targetIedName, Set<FcdaForDataSetsCreation> allowedFcdas) {
+        checkFcdaInitDataPresence(allowedFcdas);
+        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
```

Comment:
Is it the good time to get rid of these "adapter" ?
EveryWhere

Reply:
![image](https://github.com/com-pas/compas-sct/assets/105049157/eb05957d-439c-4a6c-9e76-454844e3c867)

Reply:
It's so funny picture.

Reply:
I just include the refactoring that was specified.. https://gopro-tickets.rte-france.com/browse/RSR-681

Reply:
Hello @gleizesDor, @massifben ,
I think there are specific tickets for dealing with adapters.

Reply:
ok next time